Add docs to run the new benchmark suite locally (#12614)
openxla/iree: build_tools/python/e2e_test_framework/device_specs/README.md

# IREE Benchmark Device Specs

This direcotry contains the specifications of all target devices we run

> benchmarks. Definitions can be found in `*_sepcs.py`.

Reviewer: Typo? Also wording suggestion

```suggestion
benchmarks. Definitions can be found in the `*_specs.py` files.
```

## Adding a new device spec

> 1.  Register an unique device spec ID

Reviewer: ```suggestion
1.  Register a unique device spec ID
```

It's based on sound not spelling :-)

>     in[build_tools/python/e2e_test_framework/unique_ids.py](/build_tools/python/e2e_test_framework/unique_ids.py).

Reviewer: Typo

```suggestion
    in [build_tools/python/e2e_test_framework/unique_ids.py](/build_tools/python/e2e_test_framework/unique_ids.py).
```
